### PR TITLE
Convert Pill to TypeScript

### PR DESCRIPTION
**Pull request by lavelle**

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // Silence errors from libraries with no type defintions provided
 declare module 'intersection-observer';
 declare module 'object-fit-images';
+// TODO(giles): remove once we add types to tp-icons
+declare module '@thumbtack/thumbprint-icons';
```
Comment (danoc) [inline typings/index.d.ts]: Why is this needed if `icon` is `React.ReactNode`?

Comment (lavelle) [reply typings/index.d.ts]: it's imported by the test suite, which throws an error right now

Comment (danoc) [reply typings/index.d.ts]: Ah, ok. Makes sense!

**Comment by danoc** [comment]
The Netlify build is failing because of a change to tokens. I'll fix that right now.

**Comment by danoc** [comment]
Fix for the build issue is here:
https://github.com/thumbtack/thumbprint-tokens/pull/27
